static/src/js/containers/SavedProjectsContainer/__tests__/SavedProjectsContainer.test.jsx

```diff
@@ -0,0 +1,220 @@
+    const spy = jest.spyOn(actions, 'changePath')
+
+    mapDispatchToProps(dispatch).onChangePath('path')
+
+    expect(spy).toHaveBeenCalledWith('path')
+  })
+
+  test('dispatchHandleError calls actions.handleError', () => {
+    const dispatch = jest.fn()
+    const spy = jest.spyOn(actions, 'handleError')
+    const errorConfig = { error: 'test-error' }
+
+    mapDispatchToProps(dispatch).dispatchHandleError(errorConfig)
+
+    expect(spy).toHaveBeenCalledWith(errorConfig)
+  })
+})
+
+describe('mapStateToProps', () => {
+  test('returns the correct state', () => {
+    getEarthdataEnvironment.mockReturnValue('prod-env')
+    const store = {
+      authToken: 'mock-token'
+    }
+
+    expect(mapStateToProps(store)).toEqual({
+      authToken: 'mock-token',
+      earthdataEnvironment: 'prod-env'
+    })
+  })
+})
+
+describe('SavedProjectsContainer', () => {
+  describe('When the component mounts with a valid authToken', () => {
+    test('fetches the projects and displays them', async () => {
+      const mockProjects = [
+        {
+          id: '123',
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      setup()
+
+      expect(await screen.findByText('Project 1')).toBeInTheDocument()
+      expect(screen.getByRole('button', { name: /remove project/i })).toBeInTheDocument()
+    })
+  })
+
+  describe('When the authToken is null', () => {
+    test('renders no projects', async () => {
+      setup({
+        overrideProps: {
+          authToken: null
+        }
+      })
+
+      // There should never be a remove button if no projects loaded
+      await waitFor(() => {
+        expect(screen.queryByRole('button', { name: /remove project/i })).toBeNull()
+      })
+    })
+  })
+
+  describe('When deleting a project', () => {
+    const projectId = '123'
+
+    test('successfully deletes the project and shows a success toast', async () => {
+      const mockProjects = [
+        {
+          id: projectId,
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      nock(/localhost/)
+        .delete(`/projects/${projectId}`)
+        .reply(204)
+
+      window.confirm = jest.fn(() => true)
+
+      setup()
+
+      const removeBtn = await screen.findByRole('button', { name: /remove project/i })
+
+      // eslint-disable-next-line testing-library/no-unnecessary-act
+      await act(async () => {
+        await userEvent.click(removeBtn)
+      })
+
+      await waitFor(() => {
+        expect(addToast).toHaveBeenCalledWith('Project removed', {
+          appearance: 'success',
+          autoDismiss: true
+        })
+      })
+
+      expect(nock.isDone()).toBe(true)
```

Comment:
We don't do this anywhere else in the repo, so it isn't all that useful here.

After looking at what `isDone()` is doing I think it could be useful, but in a much more impactful way.

You can add things that run globally in `jestConfigs/test-env.js`. This block will fail a test if any nock mocks are still pending (haven't been used by the test)

```
afterEach(() => {
  if (!nock.isDone()) {
    console.error('Pending mocks:', nock.pendingMocks())
  }
})
```

This is a useful change, but it results in 37 failed tests, which I think is outside the scope of this ticket